phoenix-core/src/main/java/org/apache/phoenix/compile/WhereOptimizer.java

```diff
@@ -2060,10 +2084,13 @@ public Integer getScale() {
 
                                 @Override
                                 public SortOrder getSortOrder() {
-                                    // The parts of the RVC have already been converted
-                                    // to ascending, so we don't need to consider the
-                                    // childPart sort order.
-                                    return SortOrder.ASC;
+                                    //WARNING HACK: Handle the different paths for InList vs Normal Comparison
```

Comment:
Please file and reference a JIRA to fix this and update this comment.

Reply:
Made https://issues.apache.org/jira/browse/PHOENIX-4969 and will update the comment.